modules/spa-s3-cloudfront/lambda_edge.tf

```diff
@@ -0,0 +1,133 @@
+locals {
+  lambda_edge_redirect_404_enabled = local.enabled && var.lambda_edge_redirect_404_enabled
+
+  cloudfront_lambda_function_association = concat(var.cloudfront_lambda_function_association, module.lambda_edge.lambda_function_association)
+}
+
+# See CHANGELOG for PR #978:
+# https://github.com/cloudposse/terraform-aws-components/pull/978
+#
+# Lambda@Edge was moved from submodules to this file
+moved {
+  from = module.lambda-edge-preview.module.lambda_edge.aws_lambda_function.default["origin_request"]
+  to   = module.lambda_edge.aws_lambda_function.default["origin_request"]
+}
+moved {
+  from = module.lambda_edge_redirect_404.module.lambda_edge.aws_lambda_function.default["origin_response"]
+  to   = module.lambda_edge.aws_lambda_function.default["origin_response"]
+}
+moved {
+  from = module.lambda_edge_redirect_404.module.lambda_edge.aws_lambda_function.default["viewer_request"]
+  to   = module.lambda_edge.aws_lambda_function.default["viewer_request"]
+}
+
+module "lambda_edge_functions" {
+  source  = "cloudposse/config/yaml//modules/deepmerge"
+  version = "1.0.2"
+
+  count = local.enabled ? 1 : 0
+
+  maps = [
+    local.preview_environment_enabled ? {
+      origin_request = {
+        source = [{
+          content  = <<-EOT
+          exports.handler = (event, context, callback) => {
+              const site_fqdn = "${local.site_fqdn}";
+
+              const { request } = event.Records[0].cf;
+              const default_prefix = "";
+
+              console.log('request:' + JSON.stringify(request));
+              const host = request.headers['x-forwarded-host'][0].value;
+              if (host == site_fqdn) {
+                request.origin.custom.path = default_prefix; // use default prefix if there is no subdomain
+              } else {
+                const subdomain = host.replace('.' + site_fqdn, '');
+                request.origin.custom.path = `/$${subdomain}`; // use preview prefix
```

Comment:
the double `$$` results in a single `$`, since this is passed through templating with Terraform

![CleanShot 2024-02-08 at 08 47 15](https://github.com/cloudposse/terraform-aws-components/assets/14060048/2597d4c3-99b3-42ee-baca-f4aca96b55a7)